redbot/core/settings_caches.py

```diff
@@ -210,26 +217,30 @@ def __init__(self, config: Config):
     async def add_to_blacklist(self, guild: Optional[discord.Guild], role_or_user: List[int]):
         gid: Optional[int] = guild.id if guild else None
         role_or_user = role_or_user or []
-        if not isinstance(role_or_user, list) and not all(
-            isinstance(r_or_u, str) for r_or_u in role_or_user
+        if not isinstance(role_or_user, list) and all(
+            isinstance(r_or_u, int) for r_or_u in role_or_user
         ):
             raise TypeError("Blacklisted objects must be a list of ints")
         if gid is None:
+            if gid not in self._cached_blacklist:
+                self._cached_blacklist[gid] = await self._config.blacklist()
             for obj_id in role_or_user:
-                if obj_id not in self._cached_blacklist:
+                if obj_id not in self._cached_blacklist[gid]:
                     self._cached_blacklist[gid].append(obj_id)
                     async with self._config.blacklist() as curr_list:
                         curr_list.append(obj_id)
         else:
+            if gid not in self._cached_blacklist:
+                self._cached_blacklist[gid] = self._config.guild_from_id(gid).blacklist()
             for obj_id in role_or_user:
-                if obj_id not in self._cached_blacklist:
+                if obj_id not in self._cached_blacklist[gid]:
                     self._cached_blacklist[gid].append(obj_id)
                     async with self._config.guild_from_id(gid).blacklist() as curr_list:
                         curr_list.append(obj_id)
 
     async def clear_blacklist(self, guild: Optional[discord.Guild] = None):
         gid: Optional[int] = guild.id if guild else None
-        self._cached_whitelist[gid]
+        self._cached_blacklist[gid]
```

Comment:
I missed this on first review:
```suggestion
        self._cached_blacklist[gid] = []
```